go/teams/loader.go

```diff
@@ -532,19 +550,29 @@ func (l *TeamLoader) load2InnerLockedRetry(ctx context.Context, arg load2ArgT) (
 		tmp := ret.DeepCopy()
 		ret = &tmp
 	} else {
-		l.G().Log.CDebugf(ctx, "TeamLoader not using snapshot")
+		mctx.Debug("TeamLoader not using snapshot")
 	}
 
 	tracer.Stage("merkle")
 	var lastSeqno keybase1.Seqno
 	var lastLinkID keybase1.LinkID
+
 	if (ret == nil) || repoll {
-		l.G().Log.CDebugf(ctx, "TeamLoader looking up merkle leaf (force:%v)", arg.forceRepoll)
+		mctx.Debug("TeamLoader looking up merkle leaf (force:%v)", arg.forceRepoll)
+		// Request also, without an additional RTT, freshness information about the hidden chain;
+		// we're going to send up information we know about the visible and hidden chains to both prove
+		// membership and show what we know about.
+		harg, err := hiddenPackage.MerkleLoadArg(mctx)
+		if err != nil {
+			return nil, err
+		}
+		var hif bool // hidden is fresh
```

Comment:
The fact that this is coming from the merkle client is just incidental right, since hidden seqno is not coming from merkle tree? Plugging it into the blind merkle tree serverside might result in staleness since the blind merkle tree is slower to update?

Reply:
it's to save a round-trip